File: android/src/main/java/com/getcapacitor/community/classes/options/FingerprintOptions.java
package com.getcapacitor.community.classes.options;

import static com.getcapacitor.community.BridgefyHelper.makeUUID;

import androidx.annotation.Nullable;
import com.getcapacitor.PluginCall;
import java.util.UUID;

public class FingerprintOptions {

    @Nullable
    private UUID userID;

    public FingerprintOptions(PluginCall call) {
        @Nullable
        String userID = call.getString("userID");
        this.setUserID(userID);
    }

    private void setUserID(@Nullable String userID) {
        this.userID = makeUUID(userID);
    }

    @Nullable
    public UUID getUserID() {
        return userID;
    }
}

File: ios/Sources/BridgefyPlugin/Classes/Options/FingerprintOptions.swift
import Foundation
import Capacitor

@objc public class FingerprintOptions: NSObject {
    private var userID: UUID?

    init(_ call: CAPPluginCall) {
        super.init()

        if let userID = call.getString("userID") {
            self.setUserID(userID)
        }
    }

    func setUserID(_ userID: String?) {
        self.userID = Helper.makeUUID(userID)
    }

    func getUserID() -> UUID? {
        return self.userID
    }
}
